matrix_3x3_3.0.cpp

```diff
@@ -0,0 +1,343 @@
+	A.height = height;
+	FILE* fp;
+	fopen_s(&fp, name, "r");
+	for (int row_num = 0; row_num < A.height; row_num++)
+	{
+		for (int col_num = 0; col_num < A.width; col_num++)
+		{
+			fscanf_s(fp, "%f", &A.data[row_num][col_num]);
+		}
+	}
+	fclose(fp);
+	return A;
+}
+
+void WRITE_to_file(void* A, int operation)
+{
+	FILE* fp;
+
+	if (sizeof(A) == 4)
+	{
+		fopen_s(&fp, "mat_res.txt", "a+");
+		{
+			fprintf(fp, "%c", "\n");
+			fprintf(fp, "%s", "Det A:");
+			fprintf(fp, "%f", *(float*)A);
+		}
+		fclose(fp);
+	}
+
+	else
+	{
+		switch (operation)
+		{
+		case 1:
+			fopen_s(&fp, "mat_res.txt", "a+");
+			fprintf(fp, "%c", "\n");
+			fprintf(fp, "%s", "A + B");
+			fclose(fp);
+			break;
+		case 2:
+			fopen_s(&fp, "mat_res.txt", "a+");
+			fprintf(fp, "%c", "\n");
+			fprintf(fp, "%s", "A x B");
+			fclose(fp);
+			break;
+		case 3:
+			fopen_s(&fp, "mat_res.txt", "a+");
+			fprintf(fp, "%c", "\n");
+			fprintf(fp, "%s", "transposed A");
+			fclose(fp);
+			break;
+		case 4:
+			fopen_s(&fp, "mat_res.txt", "a+");
+			fprintf(fp, "%c", "\n");
+			fprintf(fp, "%s", "inverse A");
+			fclose(fp);
+			break;
+		}
+
+		fopen_s(&fp, "mat_res.txt", "a+");
+		fprintf(fp, "%c", "\n");
+		for (int row_num = 0; row_num < ((Matrix*)A)->height; row_num++)
+		{
+			for (int col_num = 0; col_num < ((Matrix*)A)->width; col_num++)
+			{
+				fprintf(fp, "%f\t", ((Matrix*)A)->data[row_num][col_num]);
+			}
+			fprintf(fp, "%c", "\n");
+		}
+		//fprintf(fp, "%c", "\n");
+		fclose(fp);
+	}
+}
+
+Matrix summ(Matrix* A, Matrix* B)
+{
+	Matrix res;
+	res.width = A->width;
+	res.height = A->height;
+	
+	for (int row_num = 0; row_num < A->height; row_num++)
+	{
+		for (int col_num = 0; col_num < A->width; col_num++)
+		{
+			res.data[row_num][col_num] = A->data[row_num][col_num] + B->data[row_num][col_num];
+		}
+	}
+	return res;
+}
+
+Matrix multiplication(Matrix* A, Matrix* B)
+{
+	Matrix res;
+	res.width = B->width;
+	res.height = A->height;
+	for (int row_num = 0; row_num < res.height; row_num++)
+	{
+		for (int col_num = 0; col_num < res.width; col_num++)
+		{
+			res.data[row_num][col_num] = 0;
+		}
+	}
+
+	for (int row_num = 0; row_num < A->height; row_num++)
+	{
+		for (int col_num = 0; col_num < A->width; col_num++)
+		{
+			for (int k = 0; k < A->width; k++)
+			{
+				res.data[row_num][col_num] += A->data[row_num][k] * B->data[k][col_num];
+			}
+		}
+	}
+	return res;
+}
+
+Matrix transpose(Matrix* A)
+{
+	Matrix res;
+	res.width = A->width;
+	res.height = A->height;
+
+	for (int col_num = 0; col_num < res.width; col_num++) //фиксируем столбец и печатаем элементы этого столбца в строчку
+	{
+		for (int row_num = 0; row_num < res.height; row_num++)
+		{
+			res.data[row_num][col_num] = A->data[col_num][row_num];
+			//fprintf(fp, "%f", res.data[row_num][col_num]); ///вопрос: почему в файл записывает НЕ транспонированную матрицу, а исходную
+		}
+		//fprintf(fp, "%c", '\n');
+	}
+	//fprintf(fp, "%c", '\n');
+	return res;
+}
+
+float findDet(Matrix* A)

```

Comment:
"find" in name is pointless. you did not use it for "mupliplication", why do you want to use it here then?
call function matrixDeterminant()